code/client/munkilib/munkicommon.py

```diff
@@ -2772,15 +2772,33 @@ def blockingApplicationsRunning(pkginfoitem):
     return False
 
 def supports_auth_restart():
-    """Check if the machine supports an authorized
-    restart, returns True or False accordingly
-    NOTE: This does not check to see if FileVault is
-    enabled as it may return true on a machine with
-    FileVault disabled."""
-    cmd = ['/usr/bin/fdesetup', 'supportsauthrestart']
-    if subprocess.check_output(cmd).strip() == 'true':
+    """Check if FileVault is enabled then checks
+    if an Authorized Restart is supported, returns True
+    or False accordingly.
+    """
+    display_debug1('Checking if FileVault is Enabled...')
+    active_cmd = ['/usr/bin/fdesetup','isactive']
+    try:
+        is_active = subprocess.check_output(active_cmd).strip()
+    except subprocess.CalledProcessError:
+        display_warning('FileVault appears to be Disabled...')
+        return False
+    display_debug1(
+        'Checking if FileVault can perform an Authorized Restart...')
+    support_cmd = ['/usr/bin/fdesetup', 'supportsauthrestart']
+    try:
+        is_supported = subprocess.check_output(support_cmd).strip()
+    except subprocess.CalledProcessError:
+        display_warning(
+            'FileVault appears to not support an Authorized Restart...')
```

Comment:
Or more accurately/specifically -- the current machine/hardware does not support authrestart.